color_generator/models/base.py

```diff
@@ -1,71 +1,121 @@
 """Model class, to be extended by specific types of models."""
-# pylint: disable=missing-function-docstring
 from pathlib import Path
-from typing import Callable, Dict, Optional
-
-from tensorflow.keras.models import Model as KerasModel
-from tensorflow.keras.optimizers import Adam
-
-
-DIRNAME = Path(__file__).parents[1].resolve() / "weights"
+import torch.optim as optim
+import torch.nn as nn
+import torch
+from .early_stopping import EarlyStopping
 
 
 class Model:
     """Base class, to be subclassed by predictors for specific type of data."""
 
-    def __init__(
-        self,
-        dataset_cls: type,
-        network_fn: Callable[..., KerasModel],
-        dataset_args: Dict = None,
-        network_args: Dict = None,
-    ):
-        self.name = f"{self.__class__.__name__}_{dataset_cls.__name__}_{network_fn.__name__}"
-
-        if dataset_args is None:
-            dataset_args = {}
-        self.data = dataset_cls(**dataset_args)
+    def __init__(self, dataloaders, network_fn, device="cpu"):
 
-        if network_args is None:
-            network_args = {}
-        self.network = network_fn(**network_args)
-        self.network.summary()
+        self.device = device
+        self._dataloaders = dataloaders
+        self.dataset = self._dataloaders._dataset
+        self.network = network_fn.to(self.device)
 
-    @property
-    def weights_filename(self) -> str:
-        DIRNAME.mkdir(parents=True, exist_ok=True)
-        return str(DIRNAME / f"{self.name}_weights.h5")
-
-    def fit(
-        self, dataset, batch_size: int = 32, epochs: int = 10, callbacks: list = None,
-    ):
-        if callbacks is None:
-            callbacks = []
-
-        self.network.compile(loss=self.loss(), optimizer=self.optimizer())
-
-        self.network.fit(
-            dataset,
-            epochs=epochs,
-            callbacks=callbacks,
-            # validation_data=test_sequence,
-            use_multiprocessing=False,
-            workers=1,
-            shuffle=True,
+        self.name = f"{self.__class__.__name__}_{self.dataset.__class__.__name__}_{self.network.__class__.__name__}"
+        self._early_stopping = EarlyStopping(
+            patience=1, verbose=True, delta=0.001, path="early_stopping_checkpoint.pt"
```

Comment:
Early stopping hyperparameters could be parameterized instead of hardcoding them